code/ConvertToStandardPath_SubmarineCable.py

```diff
@@ -295,38 +285,95 @@ def calculate_distance_between_cable_landing_points(wktToCities, cableIdToWkt):
+                    submarine_standard_paths.append(((start_city_name, start_city_state, start_city_country, 
+                               end_city_name, end_city_state, end_city_country, 
+                               shortest_path_length, list_to_linestring(shortest_path))))
                 except NetworkXNoPath:
                     continue
                     # print(
-                    #     f"city {city_pair[0][0]}, {start_city}, city {city_pair[1][0]}, {end_city} on {cable_id} does not have shortest_path_length")
+                    #     f"city {start_city_name}, {start_city_state}, {start_city_country}, {start_city_coord}, city {end_city_name}, {end_city_state}, {end_city_country}, {end_city_coord} on {cable_id} does not have shortest_path_length")
             else:
-                if not start_city in graph:
-                    fail_cities.add(start_city)
-                    print(f"city {city_pair[0][0]}, {start_city} not exist in graph for {cable_id}.")
+                if not start_city_coord in graph:
+                    fail_cities.add(start_city_coord)
+                    print(f"city {start_city_name}, {start_city_coord} not exist in graph for {cable_id}.")
                 else:
-                    fail_cities.add(end_city)
-                    print(f"city {city_pair[1][0]}, {end_city} not exist in graph for {cable_id}.")
+                    fail_cities.add(end_city_coord)
+                    print(f"city {end_city_name}, {end_city_coord} not exist in graph for {cable_id}.")
+    return submarine_standard_paths
 
+def insert_submarine_standard_paths_to_database(db_file, submarine_standard_paths):
+    conn = sqlite3.connect(db_file)
+    cursor = conn.cursor()
+
+    # Check if the table exists
+    cursor.execute("SELECT name FROM sqlite_master WHERE type='table' AND name='submarine_standard_paths';")
+    table_exists = cursor.fetchone()
+
+    # If the table exists, drop it
+    if table_exists:
+        cursor.execute("DROP TABLE submarine_standard_paths;")
+
+    # print("Table dropped")
+
+    cursor.execute("CREATE TABLE submarine_standard_paths ( \
+        from_city TEXT, \
+        from_state TEXT, \
+        from_country TEXT, \
+        to_city TEXT, \
+        to_state TEXT, \
+        to_country TEXT, \
+        distance_km REAL, \
+        path_wkt TEXT \
+    );")
+
+    # print("Table created.")
+
+    for submarine_standard_path in submarine_standard_paths:
+        # Execute insert query
+                cursor.execute("INSERT INTO submarine_standard_paths (from_city, from_state, from_country, to_city, to_state, to_country, distance_km, path_wkt) VALUES (?, ?, ?, ?, ?, ?, ?, ?)", submarine_standard_path)
+                # print("One data inserted")
+                
+    conn.commit()
+    conn.close()
+
+def submarine_standard_path_table_tester(db_file):
```

Comment:
This is likely not need, but if you want to leave the code there, use a name that describes what this does.

This looks to me a `get_all_submarine_standard_paths()`. I mean that way it can be reused for that purpose. Calling it `..._tester` when it's not quite testing anything other than a standard get all is not ideal in my opinion.

Reply:
I understand, I will pay more attention to names for function that used for verification.